analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicRuleCheck.cs

```diff
@@ -30,6 +31,8 @@ public abstract class SymbolicRuleCheck : SymbolicCheck
         protected SonarAnalysisContext SonarContext { get; private set; }
         protected SyntaxNodeAnalysisContext NodeContext { get; private set; }
 
+        private readonly HashSet<Location> reportedDiagnostics = new();
```

Comment:
Let's make `NodeContext` also `private readonly`. That will enforce that all rules will use the new `ReportIssue` methods.

We will need to expose `protected SyntaxNode Node => context.node;`